spec/spec.tex

```diff
@@ -829,6 +829,15 @@ \subsubsection{Read Under Write Behaviour}
 
 If the read-under-write flag is set to \verb|undefined|, then the value held by the read port after the appropriate read latency is undefined.
 
+
+For the purpose of defining such collisions, an ``active write port'' is either a write port or readwrite port that is used to initiate a write operation on a given clock edge.
+An ``active read port'' is a read port or readwrite port that is used to initiate a read operation on a given clock edge.
+Each operation is defined to be ``active'' for the number of cycles set by its corresponding latency, starting from the cycle where its inputs were provided to its associated port.
+Note that this excludes combinational reads, which are simply modeled as combinationally selecting from stored values
+
+For memories with independently clocked ports, a collision between a read operation and a write operation with independent clocks is defined to occur when the address of an active write port and the address of an active read port are the same for overlapping clock periods, or when any portion of a read operation overlaps part of a write operation with a matching addresses.
+In such cases, the data that is read out of the read port is undefined.
```

Comment:
Could the data that is written also be undefined (i.e. the memory is corrupted at the addressed location)? Do we want to allow that behavior/

Reply:
Oops, I deleted my comments. This would have the effect of making read operations have side effects, which would be bad. There's no reason I can see to make the contents of memory undefined.